src/Sarif.Driver/OrderedFileSpecifier.cs

```diff
@@ -103,10 +103,18 @@ private IEnumerable<IEnumeratedArtifact> EnumeratedArtifacts()
             }
             else
             {
-                WriteFilesInDirectoryToChannel(directory, filesToProcessChannel, filter, new SortedSet<string>(StringComparer.Ordinal));
-
-                filesToProcessChannel.Writer.Complete();
-                directoryEnumerationTask = Task.CompletedTask;
+                directoryEnumerationTask = Task.Run(() =>
+                {
+                    try
+                    {
+                        WriteFilesInDirectoryToChannel(directory, filesToProcessChannel, filter, new SortedSet<string>(StringComparer.Ordinal));
+                    }
+                    finally
+                    {
+                        filesToProcessChannel.Writer.Complete();
+                        directoryEnumerationTask = Task.CompletedTask;
```

Comment:
Is setting this Task accomplishing anything?  I don't think it should.  Can be safely removed, I think.

Once the finally blocks are doing the same thing, I think you can simplify the factoring by checking if(recurse) within the Task.Run() delegate, and use the same finally block for both conditions.

Good change!

Reply:
was thinking the same, fixed.